config/application.rb

```diff
@@ -66,5 +66,10 @@ class Application < Rails::Application
         end
       end
     end
+
+    #Setting invoked exceptions application for custom 404 error pages
+    require Rails.root.join("lib/not_found_exception")
+    config.exceptions_app = NotFoundException.new(Rails.public_path)
```

Comment:
This can actually be done with this line.

```ruby
config.exceptions_app = self.routes
```

So the entire lib file is uneeded.